************* Module django.contrib.admin.actions
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\actions.py:15: [R0914(too-many-locals), delete_selected] Too many local variables (16/15)
************* Module django.contrib.admin.checks
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:814: [C0330(bad-continuation), ] Wrong continued indentation (remove 2 spaces).
                    obj.list_display_links is not None):
                  | ^
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:1: [C0302(too-many-lines), ] Too many lines in module (986/500)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:46: [W0703(broad-except), check_dependencies] Catching too general exception Exception
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:90: [R1705(no-else-return), BaseModelAdminChecks._check_raw_id_fields] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:109: [R1705(no-else-return), BaseModelAdminChecks._check_raw_id_fields_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:151: [R1705(no-else-return), BaseModelAdminChecks._check_fieldsets] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:161: [R0911(too-many-return-statements), BaseModelAdminChecks._check_fieldsets_item] Too many return statements (7/6)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:201: [R1705(no-else-return), BaseModelAdminChecks._check_field_spec] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:223: [R1705(no-else-return), BaseModelAdminChecks._check_field_spec_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:240: [R1705(no-else-return), BaseModelAdminChecks._check_exclude] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:258: [R1705(no-else-return), BaseModelAdminChecks._check_form] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:267: [R1705(no-else-return), BaseModelAdminChecks._check_filter_vertical] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:280: [R1705(no-else-return), BaseModelAdminChecks._check_filter_horizontal] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:300: [R1705(no-else-return), BaseModelAdminChecks._check_filter_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:308: [R1705(no-else-return), BaseModelAdminChecks._check_radio_fields] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:329: [R1705(no-else-return), BaseModelAdminChecks._check_radio_fields_key] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:348: [R1705(no-else-return), BaseModelAdminChecks._check_radio_fields_value] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:360: [R1705(no-else-return), BaseModelAdminChecks._check_view_on_site_url] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:361: [R1705(no-else-return), BaseModelAdminChecks._check_view_on_site_url] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:378: [R1705(no-else-return), BaseModelAdminChecks._check_prepopulated_fields] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:400: [R1705(no-else-return), BaseModelAdminChecks._check_prepopulated_fields_key] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:416: [R1705(no-else-return), BaseModelAdminChecks._check_prepopulated_fields_value] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:439: [R1705(no-else-return), BaseModelAdminChecks._check_ordering] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:483: [R1705(no-else-return), BaseModelAdminChecks._check_readonly_fields] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:538: [R1705(no-else-return), ModelAdminChecks._check_save_as] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:547: [R1705(no-else-return), ModelAdminChecks._check_save_on_top] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:556: [R1705(no-else-return), ModelAdminChecks._check_inlines] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:570: [R1705(no-else-return), ModelAdminChecks._check_inlines_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:595: [R1705(no-else-return), ModelAdminChecks._check_list_display] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:618: [R1705(no-else-return), ModelAdminChecks._check_list_display_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:603: [R0911(too-many-return-statements), ModelAdminChecks._check_list_display_item] Too many return statements (7/6)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:678: [R1705(no-else-return), ModelAdminChecks._check_list_display_links_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:692: [R1705(no-else-return), ModelAdminChecks._check_list_filter] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:713: [R1705(no-else-return), ModelAdminChecks._check_list_filter_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:730: [R1705(no-else-return), ModelAdminChecks._check_list_filter_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:700: [R0911(too-many-return-statements), ModelAdminChecks._check_list_filter_item] Too many return statements (7/6)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:755: [R1705(no-else-return), ModelAdminChecks._check_list_select_related] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:763: [R1705(no-else-return), ModelAdminChecks._check_list_per_page] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:771: [R1705(no-else-return), ModelAdminChecks._check_list_max_show_all] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:780: [R1705(no-else-return), ModelAdminChecks._check_list_editable] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:794: [R1705(no-else-return), ModelAdminChecks._check_list_editable_item] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:841: [R1705(no-else-return), ModelAdminChecks._check_search_fields] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:864: [R1705(no-else-return), ModelAdminChecks._check_date_hierarchy] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:872: [W0221(arguments-differ), InlineModelAdminChecks.check] Parameters differ from overridden 'check' method
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:898: [R1705(no-else-return), InlineModelAdminChecks._check_exclude_of_parent_model] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:923: [R1705(no-else-return), InlineModelAdminChecks._check_extra] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:931: [R1705(no-else-return), InlineModelAdminChecks._check_max_num] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:941: [R1705(no-else-return), InlineModelAdminChecks._check_min_num] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:951: [R1705(no-else-return), InlineModelAdminChecks._check_formset] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:957: [W0622(redefined-builtin), must_be] Redefining built-in 'type'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:957: [W0622(redefined-builtin), must_be] Redefining built-in 'id'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:967: [W0622(redefined-builtin), must_inherit_from] Redefining built-in 'id'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\checks.py:977: [W0622(redefined-builtin), refer_to_missing_field] Redefining built-in 'id'
************* Module django.contrib.admin.filters
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\filters.py:117: [W0223(abstract-method), FieldListFilter] Method 'choices' is abstract in class 'ListFilter' but is not overridden
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\filters.py:117: [W0223(abstract-method), FieldListFilter] Method 'expected_parameters' is abstract in class 'ListFilter' but is not overridden
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\filters.py:163: [R0902(too-many-instance-attributes), RelatedFieldListFilter] Too many instance attributes (8/7)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\filters.py:399: [W0612(unused-variable), AllValuesFieldListFilter.__init__] Unused variable 'reverse_path'
************* Module django.contrib.admin.helpers
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\helpers.py:280: [W0612(unused-variable), InlineAdminFormSet.fields] Unused variable 'i'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\helpers.py:367: [R1705(no-else-return), InlineAdminForm.fk_field] Unnecessary "else" after "return"
************* Module django.contrib.admin.models
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\models.py:99: [R1705(no-else-return), LogEntry.get_change_message] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\models.py:99: [E1136(unsubscriptable-object), LogEntry.get_change_message] Value 'self.change_message' is unsubscriptable
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\models.py:136: [E1101(no-member), LogEntry.get_edited_object] Instance of 'ForeignKey' has no 'get_object_for_this_type' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\models.py:143: [E1101(no-member), LogEntry.get_admin_url] Instance of 'ForeignKey' has no 'app_label' member
************* Module django.contrib.admin.options
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:330: [W0511(fixme), ] TODO: this should be handled by some parameter to the ChangeList.
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1017: [W0511(fixme), ] FIXME - this should check if form or formsets have a FileField,
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:246: [C0325(superfluous-parens), ] Unnecessary parens after 'in' keyword
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1: [C0302(too-many-lines), ] Too many lines in module (2002/500)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:140: [R1701(consider-merging-isinstance), BaseModelAdmin.formfield_for_dbfield] Consider merging these isinstance calls to isinstance(db_field, (models.ForeignKey, models.ManyToManyField))
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:159: [E1101(no-member), BaseModelAdmin.formfield_for_dbfield] Instance of 'BaseModelAdmin' has no 'admin_site' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:168: [E1101(no-member), BaseModelAdmin.formfield_for_dbfield] Instance of 'BaseModelAdmin' has no 'admin_site' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:207: [E1101(no-member), BaseModelAdmin.get_field_queryset] Instance of 'BaseModelAdmin' has no 'admin_site' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:220: [E1101(no-member), BaseModelAdmin.formfield_for_foreignkey] Instance of 'BaseModelAdmin' has no 'admin_site' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:245: [E1101(no-member), BaseModelAdmin.formfield_for_manytomany] Instance of 'BaseModelAdmin' has no 'admin_site' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:269: [E1102(not-callable), BaseModelAdmin.get_view_on_site_url] self.view_on_site is not callable
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:284: [E1101(no-member), BaseModelAdmin.get_empty_value_display] Instance of 'BaseModelAdmin' has no 'admin_site' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:329: [E1101(no-member), BaseModelAdmin.get_queryset] Instance of 'BaseModelAdmin' has no 'model' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:339: [E1101(no-member), BaseModelAdmin.lookup_allowed] Instance of 'BaseModelAdmin' has no 'model' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:377: [E1101(no-member), BaseModelAdmin.lookup_allowed] Instance of 'BaseModelAdmin' has no 'date_hierarchy' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:378: [E1101(no-member), BaseModelAdmin.lookup_allowed] Instance of 'BaseModelAdmin' has no 'list_filter' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:336: [R0912(too-many-branches), BaseModelAdmin.lookup_allowed] Too many branches (13/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:392: [E1101(no-member), BaseModelAdmin.to_field_allowed] Instance of 'BaseModelAdmin' has no 'model' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:413: [E1101(no-member), BaseModelAdmin.to_field_allowed] Instance of 'BaseModelAdmin' has no 'admin_site' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:437: [E1101(no-member), BaseModelAdmin.has_add_permission] Instance of 'BaseModelAdmin' has no 'opts' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:452: [E1101(no-member), BaseModelAdmin.has_change_permission] Instance of 'BaseModelAdmin' has no 'opts' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:467: [E1101(no-member), BaseModelAdmin.has_delete_permission] Instance of 'BaseModelAdmin' has no 'opts' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:482: [E1101(no-member), BaseModelAdmin.has_module_permission] Instance of 'BaseModelAdmin' has no 'opts' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:631: [E1101(no-member), ModelAdmin.get_form] Class 'ModelForm' has no 'declared_fields' member
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:713: [W0622(redefined-builtin), ModelAdmin.log_addition] Redefining built-in 'object'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:729: [W0622(redefined-builtin), ModelAdmin.log_change] Redefining built-in 'object'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:745: [W0622(redefined-builtin), ModelAdmin.log_deletion] Redefining built-in 'object'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:805: [W0102(dangerous-default-value), ModelAdmin.get_action_choices] Dangerous default value BLANK_CHOICE_DASH (builtins.list) as argument
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:811: [W0612(unused-variable), ModelAdmin.get_action_choices] Unused variable 'func'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:859: [R1705(no-else-return), ModelAdmin.get_list_display_links] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:893: [R1705(no-else-return), ModelAdmin.get_search_results.construct_search] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1042: [R0914(too-many-locals), ModelAdmin.response_add] Too many local variables (16/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1155: [R1705(no-else-return), ModelAdmin.response_change] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1291: [R1705(no-else-return), ModelAdmin.response_action] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1410: [R0914(too-many-locals), ModelAdmin._changeform_view] Too many local variables (23/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1451: [R1705(no-else-return), ModelAdmin._changeform_view] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1410: [R0912(too-many-branches), ModelAdmin._changeform_view] Too many branches (18/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1410: [R0915(too-many-statements), ModelAdmin._changeform_view] Too many statements (55/50)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1514: [R0914(too-many-locals), ModelAdmin.changelist_view] Too many local variables (29/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1514: [R0911(too-many-return-statements), ModelAdmin.changelist_view] Too many return statements (7/6)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1514: [R0912(too-many-branches), ModelAdmin.changelist_view] Too many branches (26/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1514: [R0915(too-many-statements), ModelAdmin.changelist_view] Too many statements (75/50)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1692: [R0914(too-many-locals), ModelAdmin._delete_view] Too many local variables (19/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\options.py:1862: [E1135(unsupported-membership-test), InlineModelAdmin.media] Value 'self.classes' doesn't support membership test
************* Module django.contrib.admin.sites
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\sites.py:1: [C0302(too-many-lines), ] Too many lines in module (520/500)
************* Module django.contrib.admin.tests
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:33: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_until] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:33: [E0401(import-error), AdminSeleniumTestCase.wait_until] Unable to import 'selenium.webdriver.support.wait'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:47: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_for] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:47: [E0401(import-error), AdminSeleniumTestCase.wait_for] Unable to import 'selenium.webdriver.common.by'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:48: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_for] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:48: [E0401(import-error), AdminSeleniumTestCase.wait_for] Unable to import 'selenium.webdriver.support'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:58: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_for_text] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:58: [E0401(import-error), AdminSeleniumTestCase.wait_for_text] Unable to import 'selenium.webdriver.common.by'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:59: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_for_text] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:59: [E0401(import-error), AdminSeleniumTestCase.wait_for_text] Unable to import 'selenium.webdriver.support'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:70: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_for_value] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:70: [E0401(import-error), AdminSeleniumTestCase.wait_for_value] Unable to import 'selenium.webdriver.common.by'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:71: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_for_value] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:71: [E0401(import-error), AdminSeleniumTestCase.wait_for_value] Unable to import 'selenium.webdriver.support'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:82: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_until_visible] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:82: [E0401(import-error), AdminSeleniumTestCase.wait_until_visible] Unable to import 'selenium.webdriver.common.by'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:83: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_until_visible] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:83: [E0401(import-error), AdminSeleniumTestCase.wait_until_visible] Unable to import 'selenium.webdriver.support'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:93: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_until_invisible] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:93: [E0401(import-error), AdminSeleniumTestCase.wait_until_invisible] Unable to import 'selenium.webdriver.common.by'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:94: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_until_invisible] No name 'webdriver' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:94: [E0401(import-error), AdminSeleniumTestCase.wait_until_invisible] Unable to import 'selenium.webdriver.support'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:104: [E0611(no-name-in-module), AdminSeleniumTestCase.wait_page_loaded] No name 'common' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:104: [E0401(import-error), AdminSeleniumTestCase.wait_page_loaded] Unable to import 'selenium.common.exceptions'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:142: [E0611(no-name-in-module), AdminSeleniumTestCase.get_select_option] No name 'common' in module 'selenium'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:142: [E0401(import-error), AdminSeleniumTestCase.get_select_option] Unable to import 'selenium.common.exceptions'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\tests.py:162: [C1801(len-as-condition), AdminSeleniumTestCase._assertOptionsValues.<lambda>] Do not use `len(SEQUENCE)` as condition value
************* Module django.contrib.admin.utils
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:291: [C0330(bad-continuation), ] Wrong continued indentation (remove 2 spaces).
                hasattr(model_admin, name) and
              | ^
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:292: [C0330(bad-continuation), ] Wrong continued indentation (remove 2 spaces).
                not name == '__str__' and
              | ^
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:293: [C0330(bad-continuation), ] Wrong continued indentation (remove 2 spaces).
                not name == '__unicode__'):
              | ^
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:1: [C0302(too-many-lines), ] Too many lines in module (544/500)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:78: [C0200(consider-using-enumerate), quote] Consider using enumerate instead of iterating with range and len
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:91: [W0622(redefined-builtin), unquote] Redefining built-in 'list'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:121: [W0612(unused-variable), flatten_fieldsets] Unused variable 'name'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:190: [W0221(arguments-differ), NestedObjects.collect] Parameters differ from overridden 'collect' method
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:235: [W0221(arguments-differ), NestedObjects.can_fast_delete] Parameters differ from overridden 'can_fast_delete' method
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:383: [R1705(no-else-return), label_for_field] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:330: [R0912(too-many-branches), label_for_field] Too many branches (19/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:404: [R1705(no-else-return), display_for_field] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:408: [R1701(consider-merging-isinstance), display_for_field] Consider merging these isinstance calls to isinstance(field, (models.BooleanField, models.NullBooleanField))
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:401: [R0911(too-many-return-statements), display_for_field] Too many return statements (9/6)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:429: [R1705(no-else-return), display_for_value] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\utils.py:426: [R0911(too-many-return-statements), display_for_value] Too many return statements (7/6)
************* Module django.contrib.admin.widgets
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:243: [W0511(fixme), ] XXX: The UX does not support multiple selected values.
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:246: [W0511(fixme), ] XXX: The deletion UX can be confusing when dealing with cascading deletion.
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:53: [W0622(redefined-builtin), AdminDateWidget.__init__] Redefining built-in 'format'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:66: [W0622(redefined-builtin), AdminTimeWidget.__init__] Redefining built-in 'format'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:83: [W0233(non-parent-init-called), AdminSplitDateTime.__init__] __init__ method from a non direct base class 'MultiWidget' is called
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:79: [W0231(super-init-not-called), AdminSplitDateTime.__init__] __init__ method from base class 'SplitDateTimeWidget' is not called
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:114: [R1706(consider-using-ternary), url_params_from_lookup_dict] Consider using ternary ('1' if v else '0')
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:224: [R0902(too-many-instance-attributes), RelatedFieldWidgetWrapper] Too many instance attributes (9/7)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\widgets.py:231: [W0231(super-init-not-called), RelatedFieldWidgetWrapper.__init__] __init__ method from base class 'Widget' is not called
************* Module django.contrib.admin.templatetags.admin_list
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:38: [R1705(no-else-return), paginator_number] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:135: [W0622(redefined-builtin), result_headers] Redefining built-in 'sorted'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:96: [R0914(too-many-locals), result_headers] Too many local variables (19/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:193: [R1705(no-else-return), _coerce_field_name] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:200: [R0914(too-many-locals), items_for_result] Too many local variables (22/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:200: [R0912(too-many-branches), items_for_result] Too many branches (22/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:200: [R0915(too-many-statements), items_for_result] Too many statements (61/50)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:321: [W0612(unused-variable), result_hidden_fields] Unused variable 'res'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:344: [R0914(too-many-locals), date_hierarchy] Too many local variables (17/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_list.py:373: [R1705(no-else-return), date_hierarchy] Unnecessary "else" after "return"
************* Module django.contrib.admin.templatetags.admin_modify
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_modify.py:76: [W0612(unused-variable), cell_count] Unused variable 'field'
************* Module django.contrib.admin.templatetags.admin_urls
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_urls.py:5: [E0611(no-name-in-module), ] No name 'urllib' in module '_MovedItems'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\templatetags\admin_urls.py:5: [E0401(import-error), ] Unable to import 'django.utils.six.moves.urllib.parse'
************* Module django.contrib.admin.views.main
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\views\main.py:37: [R0902(too-many-instance-attributes), ChangeList] Too many instance attributes (35/7)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\views\main.py:38: [R0913(too-many-arguments), ChangeList.__init__] Too many arguments (13/10)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\views\main.py:101: [R0912(too-many-branches), ChangeList.get_filters] Too many branches (13/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\views\main.py:254: [W0612(unused-variable), ChangeList.get_ordering] Unused variable 'none'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\views\main.py:306: [W0612(unused-variable), ChangeList.get_ordering_field_columns] Unused variable 'none'
C:\Python\Python36-32\lib\site-packages\django\contrib\admin\views\main.py:354: [R1705(no-else-return), ChangeList.get_queryset] Unnecessary "else" after "return"
************* Module django.contrib.auth
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\__init__.py:88: [W1505(deprecated-method), _authenticate_with_backend] Using deprecated method getcallargs()
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\__init__.py:94: [W1505(deprecated-method), _authenticate_with_backend] Using deprecated method getcallargs()
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\__init__.py:98: [W1505(deprecated-method), _authenticate_with_backend] Using deprecated method getcallargs()
************* Module django.contrib.auth.admin
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\admin.py:134: [W0622(redefined-builtin), UserAdmin.user_change_password] Redefining built-in 'id'
************* Module django.contrib.auth.backends
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\backends.py:128: [W0221(arguments-differ), RemoteUserBackend.authenticate] Parameters differ from overridden 'authenticate' method
************* Module django.contrib.auth.base_user
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\base_user.py:65: [E1101(no-member), AbstractBaseUser.get_username] Instance of 'AbstractBaseUser' has no 'USERNAME_FIELD' member
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\base_user.py:77: [E1101(no-member), AbstractBaseUser.clean] Instance of 'AbstractBaseUser' has no 'USERNAME_FIELD' member
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\base_user.py:79: [W0221(arguments-differ), AbstractBaseUser.save] Parameters differ from overridden 'save' method
************* Module django.contrib.auth.decorators
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\decorators.py:9: [E0611(no-name-in-module), ] No name 'urllib' in module '_MovedItems'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\decorators.py:9: [E0401(import-error), ] Unable to import 'django.utils.six.moves.urllib.parse'
************* Module django.contrib.auth.forms
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\forms.py:258: [R0914(too-many-locals), PasswordResetForm.save] Too many local variables (16/15)
************* Module django.contrib.auth.hashers
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:343: [C0330(bad-continuation), ] Wrong continued indentation (remove 3 spaces).
            salt, data) = self._decode(encoded)
         |  ^
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:358: [C0330(bad-continuation), ] Wrong continued indentation (remove 3 spaces).
            salt, data) = self._decode(encoded)
         |  ^
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:1: [C0302(too-many-lines), ] Too many lines in module (653/500)
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:157: [W0622(redefined-builtin), mask_hash] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:182: [W0612(unused-variable), BasePasswordHasher._load_library] Unused variable 'name'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:258: [W0622(redefined-builtin), PBKDF2PasswordHasher.encode] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:253: [W0221(arguments-differ), PBKDF2PasswordHasher.encode] Parameters differ from overridden 'encode' method
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:263: [W0622(redefined-builtin), PBKDF2PasswordHasher.verify] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:263: [W0612(unused-variable), PBKDF2PasswordHasher.verify] Unused variable 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:269: [W0622(redefined-builtin), PBKDF2PasswordHasher.safe_summary] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:279: [W0622(redefined-builtin), PBKDF2PasswordHasher.must_update] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:279: [W0612(unused-variable), PBKDF2PasswordHasher.must_update] Unused variable 'algorithm'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:279: [W0612(unused-variable), PBKDF2PasswordHasher.must_update] Unused variable 'salt'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:279: [W0612(unused-variable), PBKDF2PasswordHasher.must_update] Unused variable 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:283: [W0622(redefined-builtin), PBKDF2PasswordHasher.harden_runtime] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:283: [W0612(unused-variable), PBKDF2PasswordHasher.harden_runtime] Unused variable 'algorithm'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:283: [W0612(unused-variable), PBKDF2PasswordHasher.harden_runtime] Unused variable 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:357: [W0612(unused-variable), Argon2PasswordHasher.must_update] Unused variable 'variety'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:358: [W0612(unused-variable), Argon2PasswordHasher.must_update] Unused variable 'salt'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:358: [W0612(unused-variable), Argon2PasswordHasher.must_update] Unused variable 'data'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:440: [W0612(unused-variable), BCryptSHA256PasswordHasher.safe_summary] Unused variable 'empty'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:440: [W0612(unused-variable), BCryptSHA256PasswordHasher.safe_summary] Unused variable 'algostr'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:451: [W0612(unused-variable), BCryptSHA256PasswordHasher.must_update] Unused variable 'algorithm'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:451: [W0612(unused-variable), BCryptSHA256PasswordHasher.must_update] Unused variable 'empty'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:451: [W0612(unused-variable), BCryptSHA256PasswordHasher.must_update] Unused variable 'algostr'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:451: [W0612(unused-variable), BCryptSHA256PasswordHasher.must_update] Unused variable 'data'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:493: [W0622(redefined-builtin), SHA1PasswordHasher.encode] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:497: [W0622(redefined-builtin), SHA1PasswordHasher.verify] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:497: [W0612(unused-variable), SHA1PasswordHasher.verify] Unused variable 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:503: [W0622(redefined-builtin), SHA1PasswordHasher.safe_summary] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:524: [W0622(redefined-builtin), MD5PasswordHasher.encode] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:528: [W0622(redefined-builtin), MD5PasswordHasher.verify] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:528: [W0612(unused-variable), MD5PasswordHasher.verify] Unused variable 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:534: [W0622(redefined-builtin), MD5PasswordHasher.safe_summary] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:562: [W0622(redefined-builtin), UnsaltedSHA1PasswordHasher.encode] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:571: [W0622(redefined-builtin), UnsaltedSHA1PasswordHasher.safe_summary] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\hashers.py:639: [W0612(unused-variable), CryptPasswordHasher.verify] Unused variable 'salt'
************* Module django.contrib.auth.models
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\models.py:83: [E1101(no-member), Permission.__str__] Instance of 'ForeignKey' has no 'app_label' member
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\models.py:88: [E1101(no-member), Permission.natural_key] Instance of 'ForeignKey' has no 'natural_key' member
************* Module django.contrib.auth.password_validation
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\password_validation.py:67: [W0621(redefined-outer-name), password_changed] Redefining name 'password_changed' from outer scope (line 59)
************* Module django.contrib.auth.tokens
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\tokens.py:31: [W0622(redefined-builtin), PasswordResetTokenGenerator.check_token] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\tokens.py:31: [W0612(unused-variable), PasswordResetTokenGenerator.check_token] Unused variable 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\tokens.py:62: [W0622(redefined-builtin), PasswordResetTokenGenerator._make_token_with_timestamp] Redefining built-in 'hash'
************* Module django.contrib.auth.views
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:1: [C0302(too-many-lines), ] Too many lines in module (631/500)
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:26: [E0611(no-name-in-module), ] No name 'urllib' in module '_MovedItems'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:26: [E0401(import-error), ] Unable to import 'django.utils.six.moves.urllib.parse'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:250: [W0622(redefined-builtin), redirect_to_login] Redefining built-in 'next'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:275: [R0913(too-many-arguments), password_reset] Too many arguments (11/10)
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:438: [W0221(arguments-differ), PasswordResetView.dispatch] Parameters differ from overridden 'dispatch' method
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:476: [W0221(arguments-differ), PasswordResetConfirmView.dispatch] Parameters differ from overridden 'dispatch' method
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:479: [W0201(attribute-defined-outside-init), PasswordResetConfirmView.dispatch] Attribute 'validlink' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:488: [W0201(attribute-defined-outside-init), PasswordResetConfirmView.dispatch] Attribute 'validlink' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:480: [W0201(attribute-defined-outside-init), PasswordResetConfirmView.dispatch] Attribute 'user' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:609: [W0221(arguments-differ), PasswordChangeView.dispatch] Parameters differ from overridden 'dispatch' method
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\views.py:630: [W0221(arguments-differ), PasswordChangeDoneView.dispatch] Parameters differ from overridden 'dispatch' method
************* Module django.contrib.auth.management
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\__init__.py:40: [R0914(too-many-locals), create_permissions] Too many local variables (16/15)
************* Module django.contrib.auth.management.commands.createsuperuser
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\commands\createsuperuser.py:115: [C0330(bad-continuation), ] Wrong continued indentation (add 21 spaces).
                    )
                    ^                    |
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\commands\createsuperuser.py:16: [W0622(redefined-builtin), ] Redefining built-in 'input'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\commands\createsuperuser.py:16: [E0401(import-error), ] Unable to import 'django.utils.six.moves'
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\commands\createsuperuser.py:65: [R0914(too-many-locals), Command.handle] Too many local variables (19/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\commands\createsuperuser.py:65: [R0912(too-many-branches), Command.handle] Too many branches (25/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\commands\createsuperuser.py:65: [R0915(too-many-statements), Command.handle] Too many statements (75/50)
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\management\commands\createsuperuser.py:62: [W0201(attribute-defined-outside-init), Command.execute] Attribute 'stdin' defined outside __init__
************* Module django.contrib.auth.migrations.0001_initial
C:\Python\Python36-32\lib\site-packages\django\contrib\auth\migrations\0001_initial.py:98: [C0330(bad-continuation), ] Wrong continued indentation (remove 1 space).
                 ),
                |^
************* Module django.contrib.contenttypes.admin
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\admin.py:30: [R1705(no-else-return), GenericInlineModelAdminChecks._check_relation] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\admin.py:30: [C1801(len-as-condition), GenericInlineModelAdminChecks._check_relation] Do not use `len(SEQUENCE)` as condition value
************* Module django.contrib.contenttypes.fields
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:1: [C0302(too-many-lines), ] Too many lines in module (657/500)
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:20: [R0902(too-many-instance-attributes), GenericForeignKey] Too many instance attributes (9/7)
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:86: [R1705(no-else-return), GenericForeignKey._check_field_name] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:129: [R1705(no-else-return), GenericForeignKey._check_content_type_field] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:160: [W0622(redefined-builtin), GenericForeignKey.get_content_type] Redefining built-in 'id'
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:199: [R1705(no-else-return), GenericForeignKey.get_prefetch_queryset.gfk_key] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:53: [W0201(attribute-defined-outside-init), GenericForeignKey.contribute_to_class] Attribute 'name' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:54: [W0201(attribute-defined-outside-init), GenericForeignKey.contribute_to_class] Attribute 'model' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:55: [W0201(attribute-defined-outside-init), GenericForeignKey.contribute_to_class] Attribute 'cache_attr' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:335: [R1705(no-else-return), GenericRelation._check_generic_foreign_key_existence] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:337: [R1705(no-else-return), GenericRelation._check_generic_foreign_key_existence] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:392: [R1705(no-else-return), GenericRelation.get_path_info] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:407: [W0221(arguments-differ), GenericRelation.contribute_to_class] Parameters differ from overridden 'contribute_to_class' method
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:437: [W0221(arguments-differ), GenericRelation.get_extra_restriction] Parameters differ from overridden 'get_extra_restriction' method
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:410: [W0201(attribute-defined-outside-init), GenericRelation.contribute_to_class] Attribute 'model' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\fields.py:484: [R0902(too-many-instance-attributes), create_generic_related_manager.GenericRelatedObjectManager] Too many instance attributes (8/7)
************* Module django.contrib.contenttypes.forms
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:20: [E1101(no-member), BaseGenericInlineFormSet.__init__] Instance of 'BaseGenericInlineFormSet' has no 'ct_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:20: [E1101(no-member), BaseGenericInlineFormSet.__init__] Instance of 'BaseGenericInlineFormSet' has no 'ct_fk_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:28: [E1101(no-member), BaseGenericInlineFormSet.__init__] Instance of 'BaseGenericInlineFormSet' has no 'ct_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:29: [E1101(no-member), BaseGenericInlineFormSet.__init__] Instance of 'BaseGenericInlineFormSet' has no 'for_concrete_model' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:30: [E1101(no-member), BaseGenericInlineFormSet.__init__] Instance of 'BaseGenericInlineFormSet' has no 'ct_fk_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:41: [E1101(no-member), BaseGenericInlineFormSet.get_default_prefix] Class 'BaseGenericInlineFormSet' has no 'ct_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:41: [E1101(no-member), BaseGenericInlineFormSet.get_default_prefix] Class 'BaseGenericInlineFormSet' has no 'ct_fk_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:44: [E1101(no-member), BaseGenericInlineFormSet.save_new] Instance of 'BaseGenericInlineFormSet' has no 'ct_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:45: [E1101(no-member), BaseGenericInlineFormSet.save_new] Instance of 'BaseGenericInlineFormSet' has no 'ct_fk_field' member
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:49: [R0913(too-many-arguments), generic_inlineformset_factory] Too many arguments (16/10)
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\forms.py:49: [R0914(too-many-locals), generic_inlineformset_factory] Too many local variables (18/15)
************* Module django.contrib.contenttypes.models
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\models.py:58: [W0612(unused-variable), ContentTypeManager.get_for_model] Unused variable 'created'
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\models.py:65: [W0621(redefined-outer-name), ContentTypeManager.get_for_models] Redefining name 'models' from outer scope (line 6)
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\models.py:109: [W0622(redefined-builtin), ContentTypeManager.get_for_id] Redefining built-in 'id'
************* Module django.contrib.contenttypes.views
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\views.py:87: [R1705(no-else-return), shortcut] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\views.py:11: [R0912(too-many-branches), shortcut] Too many branches (20/12)
************* Module django.contrib.contenttypes.management.__init__
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\management\__init__.py:65: [C0325(superfluous-parens), ] Unnecessary parens after 'if' keyword
************* Module django.contrib.contenttypes.management.commands.remove_stale_contenttypes
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\management\commands\remove_stale_contenttypes.py:7: [W0622(redefined-builtin), ] Redefining built-in 'input'
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\management\commands\remove_stale_contenttypes.py:7: [E0401(import-error), ] Unable to import 'django.utils.six.moves'
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\management\commands\remove_stale_contenttypes.py:25: [W0221(arguments-differ), Command.handle] Parameters differ from overridden 'handle' method
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\management\commands\remove_stale_contenttypes.py:25: [R0914(too-many-locals), Command.handle] Too many local variables (17/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\management\commands\remove_stale_contenttypes.py:30: [R1702(too-many-nested-blocks), Command.handle] Too many nested blocks (6/5)
C:\Python\Python36-32\lib\site-packages\django\contrib\contenttypes\management\commands\remove_stale_contenttypes.py:82: [W0221(arguments-differ), NoFastDeleteCollector.can_fast_delete] Parameters differ from overridden 'can_fast_delete' method
************* Module django.contrib.sessions.middleware
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\middleware.py:13: [W0231(super-init-not-called), SessionMiddleware.__init__] __init__ method from base class 'MiddlewareMixin' is not called
************* Module django.contrib.sessions.backends.base
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:76: [R1705(no-else-return), SessionBase.setdefault] Unnecessary "else" after "return"
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:99: [W0622(redefined-builtin), SessionBase.encode] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:106: [W0622(redefined-builtin), SessionBase.decode] Redefining built-in 'hash'
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:112: [W0703(broad-except), SessionBase.decode] Catching too general exception Exception
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:149: [W0201(attribute-defined-outside-init), SessionBase.clear] Attribute '_session_cache' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:205: [W0201(attribute-defined-outside-init), SessionBase._get_session] Attribute '_session_cache' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:207: [W0201(attribute-defined-outside-init), SessionBase._get_session] Attribute '_session_cache' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:312: [W0201(attribute-defined-outside-init), SessionBase.cycle_key] Attribute '_session_cache' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:188: [W0201(attribute-defined-outside-init), SessionBase._set_session_key] Attribute '__session_key' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\base.py:190: [W0201(attribute-defined-outside-init), SessionBase._set_session_key] Attribute '__session_key' defined outside __init__
************* Module django.contrib.sessions.backends.cache
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\cache.py:6: [W0622(redefined-builtin), ] Redefining built-in 'range'
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\cache.py:6: [E0401(import-error), ] Unable to import 'django.utils.six.moves'
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\cache.py:28: [W0703(broad-except), SessionStore.load] Catching too general exception Exception
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\cache.py:43: [W0612(unused-variable), SessionStore.create] Unused variable 'i'
************* Module django.contrib.sessions.backends.cached_db
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\cached_db.py:34: [W0703(broad-except), SessionStore.load] Catching too general exception Exception
************* Module django.contrib.sessions.backends.db
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\db.py:17: [W0235(useless-super-delegation), SessionStore.__init__] Useless super delegation in method '__init__'
************* Module django.contrib.sessions.backends.file
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\file.py:159: [W0622(redefined-builtin), SessionStore.save] Redefining built-in 'dir'
************* Module django.contrib.sessions.backends.signed_cookies
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\signed_cookies.py:22: [W0703(broad-except), SessionStore.load] Catching too general exception Exception
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\backends\signed_cookies.py:59: [W0201(attribute-defined-outside-init), SessionStore.delete] Attribute '_session_cache' defined outside __init__
************* Module django.contrib.sessions.management.commands.clearsessions
C:\Python\Python36-32\lib\site-packages\django\contrib\sessions\management\commands\clearsessions.py:13: [W0221(arguments-differ), Command.handle] Parameters differ from overridden 'handle' method
************* Module django.contrib.messages
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\__init__.py:1: [W0401(wildcard-import), ] Wildcard import django.contrib.messages.api
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\__init__.py:2: [W0401(wildcard-import), ] Wildcard import django.contrib.messages.constants
************* Module django.contrib.messages.storage.base
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\base.py:92: [W0612(unused-variable), BaseStorage._loaded_messages] Unused variable 'all_retrieved'
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\base.py:93: [W0201(attribute-defined-outside-init), BaseStorage._loaded_messages] Attribute '_loaded_data' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\base.py:170: [W0201(attribute-defined-outside-init), BaseStorage._get_level] Attribute '_level' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\base.py:183: [W0201(attribute-defined-outside-init), BaseStorage._set_level] Attribute '_level' defined outside __init__
************* Module django.contrib.messages.storage.cookie
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\cookie.py:17: [W0221(arguments-differ), MessageEncoder.default] Parameters differ from overridden 'default' method
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\cookie.py:17: [E0202(method-hidden), MessageEncoder.default] An attribute defined in json.encoder line 158 hides this method
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\cookie.py:48: [W0221(arguments-differ), MessageDecoder.decode] Parameters differ from overridden 'decode' method
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\cookie.py:94: [W0221(arguments-differ), CookieStorage._store] Parameters differ from overridden '_store' method
C:\Python\Python36-32\lib\site-packages\django\contrib\messages\storage\cookie.py:155: [W0622(redefined-builtin), CookieStorage._decode] Redefining built-in 'hash'
************* Module django.contrib.staticfiles.finders
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\finders.py:25: [W0622(redefined-builtin), BaseFinder.find] Redefining built-in 'all'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\finders.py:76: [W0622(redefined-builtin), FileSystemFinder.find] Redefining built-in 'all'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\finders.py:110: [W0612(unused-variable), FileSystemFinder.list] Unused variable 'prefix'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\finders.py:151: [W0622(redefined-builtin), AppDirectoriesFinder.find] Redefining built-in 'all'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\finders.py:199: [W0622(redefined-builtin), BaseStorageFinder.find] Redefining built-in 'all'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\finders.py:240: [W0622(redefined-builtin), find] Redefining built-in 'all'
************* Module django.contrib.staticfiles.handlers
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\handlers.py:5: [E0611(no-name-in-module), ] No name 'urllib' in module '_MovedItems'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\handlers.py:5: [E0401(import-error), ] Unable to import 'django.utils.six.moves.urllib.parse'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\handlers.py:6: [E0611(no-name-in-module), ] No name 'urllib' in module '_MovedItems'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\handlers.py:6: [E0401(import-error), ] Unable to import 'django.utils.six.moves.urllib.request'
************* Module django.contrib.staticfiles.storage
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:1: [C0302(too-many-lines), ] Too many lines in module (507/500)
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:21: [W0622(redefined-builtin), ] Redefining built-in 'range'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:21: [E0401(import-error), ] Unable to import 'django.utils.six.moves'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:22: [E0611(no-name-in-module), ] No name 'urllib' in module '_MovedItems'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:22: [E0401(import-error), ] Unable to import 'django.utils.six.moves.urllib.parse'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:251: [W0612(unused-variable), HashedFilesMixin.post_process] Unused variable 'i'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:266: [R0914(too-many-locals), HashedFilesMixin._post_process] Too many local variables (25/15)
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:266: [R0912(too-many-branches), HashedFilesMixin._post_process] Too many branches (14/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:363: [W0612(unused-variable), HashedFilesMixin.stored_name] Unused variable 'i'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\storage.py:410: [W0221(arguments-differ), ManifestFilesMixin.post_process] Parameters differ from overridden 'post_process' method
************* Module django.contrib.staticfiles.utils
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\utils.py:35: [W0622(redefined-builtin), get_files] Redefining built-in 'dir'
************* Module django.contrib.staticfiles.views
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\views.py:12: [E0611(no-name-in-module), ] No name 'urllib' in module '_MovedItems'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\views.py:12: [E0401(import-error), ] Unable to import 'django.utils.six.moves.urllib.parse'
************* Module django.contrib.staticfiles.management.commands.collectstatic
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:283: [C0325(superfluous-parens), ] Unnecessary parens after 'not' keyword
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:14: [W0622(redefined-builtin), ] Redefining built-in 'input'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:14: [E0401(import-error), ] Unable to import 'django.utils.six.moves'
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:17: [R0902(too-many-instance-attributes), Command] Too many instance attributes (13/7)
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:96: [R0912(too-many-branches), Command.collect] Too many branches (15/12)
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:159: [W0221(arguments-differ), Command.handle] Parameters differ from overridden 'handle' method
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:85: [W0201(attribute-defined-outside-init), Command.set_options] Attribute 'interactive' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:86: [W0201(attribute-defined-outside-init), Command.set_options] Attribute 'verbosity' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:87: [W0201(attribute-defined-outside-init), Command.set_options] Attribute 'symlink' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:88: [W0201(attribute-defined-outside-init), Command.set_options] Attribute 'clear' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:89: [W0201(attribute-defined-outside-init), Command.set_options] Attribute 'dry_run' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:93: [W0201(attribute-defined-outside-init), Command.set_options] Attribute 'ignore_patterns' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\collectstatic.py:94: [W0201(attribute-defined-outside-init), Command.set_options] Attribute 'post_process' defined outside __init__
************* Module django.contrib.staticfiles.management.commands.findstatic
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\findstatic.py:22: [W0221(arguments-differ), Command.handle_label] Parameters differ from overridden 'handle_label' method
C:\Python\Python36-32\lib\site-packages\django\contrib\staticfiles\management\commands\findstatic.py:37: [R1705(no-else-return), Command.handle_label] Unnecessary "else" after "return"
************* Module django_jenkins.apps
C:\Python\Python36-32\lib\site-packages\django_jenkins\apps.py:20: [W0105(pointless-string-statement), JenkinsConfig.__init__] String statement has no effect
************* Module django_jenkins.runner
C:\Python\Python36-32\lib\site-packages\django_jenkins\runner.py:103: [W0612(unused-variable), EXMLTestResult._add_tb_to_test] Unused variable 'tb'
C:\Python\Python36-32\lib\site-packages\django_jenkins\runner.py:29: [W0201(attribute-defined-outside-init), EXMLTestResult.addSuccess] Attribute 'testcase' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django_jenkins\runner.py:33: [W0201(attribute-defined-outside-init), EXMLTestResult.addFailure] Attribute 'testcase' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django_jenkins\runner.py:39: [W0201(attribute-defined-outside-init), EXMLTestResult.addError] Attribute 'testcase' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django_jenkins\runner.py:45: [W0201(attribute-defined-outside-init), EXMLTestResult.addUnexpectedSuccess] Attribute 'testcase' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django_jenkins\runner.py:51: [W0201(attribute-defined-outside-init), EXMLTestResult.addSkip] Attribute 'testcase' defined outside __init__
C:\Python\Python36-32\lib\site-packages\django_jenkins\runner.py:57: [W0201(attribute-defined-outside-init), EXMLTestResult.addExpectedFailure] Attribute 'testcase' defined outside __init__
************* Module django_jenkins.management.commands.jenkins
C:\Python\Python36-32\lib\site-packages\django_jenkins\management\commands\jenkins.py:13: [W0621(redefined-outer-name), get_runner] Redefining name 'settings' from outer scope (line 7)
C:\Python\Python36-32\lib\site-packages\django_jenkins\management\commands\jenkins.py:54: [W0106(expression-not-assigned), Command.add_arguments] Expression "(parser.add_argument('--output-dir', dest='output_dir', default='reports', help='Report files directory'), )" is assigned to nothing
C:\Python\Python36-32\lib\site-packages\django_jenkins\management\commands\jenkins.py:56: [W0106(expression-not-assigned), Command.add_arguments] Expression "(parser.add_argument('--enable-coverage', action='store_true', default=False, help='Measure code coverage'), )" is assigned to nothing
C:\Python\Python36-32\lib\site-packages\django_jenkins\management\commands\jenkins.py:59: [W0106(expression-not-assigned), Command.add_arguments] Expression "(parser.add_argument('--debug', action='store_true', dest='debug', default=False, help='Do not intercept stdout and stderr, friendly for console debuggers'), )" is assigned to nothing
C:\Python\Python36-32\lib\site-packages\django_jenkins\management\commands\jenkins.py:62: [W0106(expression-not-assigned), Command.add_arguments] Expression "(parser.add_argument('--coverage-rcfile', dest='coverage_rcfile', default='', help='Specify configuration file.'), )" is assigned to nothing
C:\Python\Python36-32\lib\site-packages\django_jenkins\management\commands\jenkins.py:66: [W0106(expression-not-assigned), Command.add_arguments] Expression "(parser.add_argument('--coverage-format', dest='coverage_format', default='xml', help='Specify coverage output formats html,xml,bin'), )" is assigned to nothing
C:\Python\Python36-32\lib\site-packages\django_jenkins\management\commands\jenkins.py:70: [W0106(expression-not-assigned), Command.add_arguments] Expression "(parser.add_argument('--coverage-exclude', action='append', default=[], dest='coverage_excludes', help='Module name to exclude'), )" is assigned to nothing
************* Module django_jenkins.tasks.run_flake8
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\run_flake8.py:11: [E0401(import-error), ] Unable to import 'flake8.api.legacy'
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\run_flake8.py:3: [C0411(wrong-import-order), ] standard import "import sys" should be placed before "import pep8"
************* Module django_jenkins.tasks.run_pep8
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\run_pep8.py:18: [W0106(expression-not-assigned), Reporter.add_arguments] Expression "(parser.add_argument('--pep8-ignore', dest='pep8-ignore', help='skip errors and warnings (e.g. E4,W)'), )" is assigned to nothing
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\run_pep8.py:30: [E0213(no-self-argument), Reporter.run.JenkinsReport.error] Method should have "self" as first argument
************* Module django_jenkins.tasks.run_pyflakes
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\run_pyflakes.py:27: [R1702(too-many-nested-blocks), Reporter.run] Too many nested blocks (6/5)
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\run_pyflakes.py:30: [W0612(unused-variable), Reporter.run] Unused variable 'dirnames'
************* Module django_jenkins.tasks.with_coverage
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\with_coverage.py:56: [C0325(superfluous-parens), ] Unnecessary parens after 'not' keyword
C:\Python\Python36-32\lib\site-packages\django_jenkins\tasks\with_coverage.py:57: [C0325(superfluous-parens), ] Unnecessary parens after 'not' keyword
************* Module animal.admin
animal\admin.py:1: [W0611(unused-import), ] Unused admin imported from django.contrib
************* Module animal.tests
animal\tests.py:20: [C0304(missing-final-newline), ] Final newline missing
animal\tests.py:20: [C0326(bad-whitespace), ] Exactly one space required after comma
        self.assertAlmostEqual(lion.__str__(),"lion")
                                             ^
************* Module animal.views
animal\views.py:1: [W0611(unused-import), ] Unused render imported from django.shortcuts
************* Module animal.migrations.__init__
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.options -> django.contrib.admin.views.main)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.helpers -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main -> django.contrib.admin.options)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.helpers -> django.contrib.admin.utils -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main -> django.contrib.admin.options)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.auth -> django.contrib.auth.models)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.options -> django.contrib.admin.templatetags.admin_urls)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.filters -> django.contrib.admin.utils -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main -> django.contrib.admin.options)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.auth.decorators -> django.contrib.auth.views)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin -> django.contrib.admin.filters -> django.contrib.admin.utils -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main -> django.contrib.admin.options -> django.contrib.admin.checks -> django.contrib.admin.sites)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.options -> django.contrib.admin.templatetags.admin_urls -> django.contrib.admin.utils -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.actions -> django.contrib.admin.helpers -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main -> django.contrib.admin.options -> django.contrib.admin.checks -> django.contrib.admin.sites)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin -> django.contrib.admin.decorators)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin -> django.contrib.admin.decorators -> django.contrib.admin.sites)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main -> django.contrib.admin.utils)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.templatetags.admin_list -> django.contrib.admin.templatetags.admin_urls -> django.contrib.admin.utils)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.checks -> django.contrib.admin.options)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.models -> django.contrib.admin.utils -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.views.main -> django.contrib.admin.options)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.helpers -> django.contrib.admin.templatetags.admin_list -> django.contrib.admin.templatetags.admin_urls -> django.contrib.admin.options)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.options -> django.contrib.admin.widgets -> django.contrib.admin.views.main)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.admin.filters -> django.contrib.admin.options)
animal\migrations\__init__.py:1: [R0401(cyclic-import), ] Cyclic import (django.contrib.sessions.backends.db -> django.contrib.sessions.models)
animal\migrations\__init__.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==django_jenkins.tasks.run_flake8:79
==django_jenkins.tasks.run_pep8:60
        output.close()

    def get_config_path(self, options):
        if options['pep8-rcfile']:
            return options['pep8-rcfile']

        rcfile = getattr(settings, 'PEP8_RCFILE', None)
        if rcfile:
            return rcfile

        if os.path.exists('tox.ini'):
            return 'tox.ini'

        if os.path.exists('setup.cfg'):
            return 'setup.cfg'
animal\migrations\__init__.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==django_jenkins.tasks.run_flake8:46
==django_jenkins.tasks.run_pep8:36
        config_file = self.get_config_path(options)
        if config_file is not None:
            pep8_options['config_file'] = config_file

        set_option(pep8_options, 'exclude', options['pep8-exclude'], config_file,
                   default=pep8.DEFAULT_EXCLUDE + ",south_migrations", split=',')

        set_option(pep8_options, 'select', options['pep8-select'], config_file, split=',')

        set_option(pep8_options, 'ignore', options['pep8-ignore'], config_file, split=',')

        set_option(pep8_options, 'max_line_length', options['pep8-max-line-length'], config_file,
                   default=pep8.MAX_LINE_LENGTH)

animal\migrations\__init__.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==django.contrib.sessions.backends.cache:14
==django.contrib.sessions.backends.cached_db:20
    cache_key_prefix = KEY_PREFIX

    def __init__(self, session_key=None):
        self._cache = caches[settings.SESSION_CACHE_ALIAS]
        super(SessionStore, self).__init__(session_key)

    @property
    def cache_key(self):
        return self.cache_key_prefix + self._get_or_create_session_key()

    def load(self):
        try:
animal\migrations\__init__.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==django_jenkins.tasks.run_csslint:1
==django_jenkins.tasks.run_scsslint:1
import os
import subprocess
import codecs

from django.conf import settings
from django_jenkins.tasks import static_files_iterator


class Reporter(object):
    def add_arguments(self, parser):
animal\migrations\__init__.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==django_jenkins.tasks.run_flake8:25
==django_jenkins.tasks.run_pep8:10
        parser.add_argument("--pep8-exclude",
                            dest="pep8-exclude",
                            help="exclude files or directories which match these "
                            "comma separated patterns (default: %s)" %
                            (pep8.DEFAULT_EXCLUDE + ",south_migrations"))
        parser.add_argument("--pep8-select", dest="pep8-select",
                            help="select errors and warnings (e.g. E,W6)")
        parser.add_argument("--pep8-ignore", dest="pep8-ignore",
animal\migrations\__init__.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==django.contrib.auth.migrations.0007_alter_validators_add_error_messages:12
==django.contrib.auth.migrations.0008_alter_user_username_max_length:12
    ]

    operations = [
        migrations.AlterField(
            model_name='user',
            name='username',
            field=models.CharField(
                error_messages={'unique': 'A user with that username already exists.'},
animal\migrations\__init__.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==django.contrib.sessions.backends.cache:43
==django.contrib.sessions.backends.file:110
            self._session_key = self._get_new_session_key()
            try:
                self.save(must_create=True)
            except CreateError:
                continue
            self.modified = True
            return

------------------------------------------------------------------
Your code has been rated at 9.13/10 (previous run: 9.13/10, +0.00)

